common/include/vecmat.h

```diff
@@ -163,7 +163,7 @@ class vm_distance_squared
 	}
 	static constexpr vm_distance_squared maximum_value()
 	{
-		return vm_distance_squared{0x7fffffffffffffff};
+		return vm_distance_squared{static_cast<fix64>(0x7fffffffffffffff)};
```

Comment:
NAK.  `0x7fffffffffffffff` is too big to fit in an `int`, so it is already `long`.  On x86_64-pc-linux-gnu, `fix64` is `long`, so this change adds a useless cast, which causes gcc to warn (due to `-Wuseless-cast`).  It looks wrong to me that clang would even consider the deleted constructor, since it should not try to bind `const fix &` to a value that is too big to fit in an `int`.

Does the below patch work for you?  `INT64_MAX` is defined in `stdint.h` to include a platform-appropriate suffix to request the right type.  For me, this code change has no effect on the generated code.

```diff
--- a/common/include/vecmat.h
+++ b/common/include/vecmat.h
@@ -163,7 +163,7 @@ public:
 	}
 	static constexpr vm_distance_squared maximum_value()
 	{
-		return vm_distance_squared{0x7fffffffffffffff};
+		return vm_distance_squared{INT64_MAX};
 	}
 	static constexpr vm_distance_squared minimum_value()
 	{
```